test/js_test.go

```diff
@@ -208,6 +218,16 @@ func TestJetStreamPublish(t *testing.T) {
 	}
 	expect(3, 3)
 
+	// JetStream Headers are case-sensitive right now,
```

Comment:
Filed separate issue for this, issue unrelated to this PR but might be good to have to avoid issues when a header changes case for some reason in one of the libraries: https://github.com/nats-io/nats-server/pull/2094